LaraApi/src-lara/lara/vitishls/VitisHls.js

```diff
@@ -0,0 +1,153 @@
+"use strict";
+
+laraImport("clava.Clava");
+laraImport("lara.util.ProcessExecutor");
+laraImport("lara.vitishls.VitisHlsReportParser");
+laraImport("lara.Tool");
+laraImport("lara.ToolUtils");
+
+class VitisHls extends Tool {
+    topFunction;
+    platform;
+    clock;
+    vitisDir;
+    vitisProjName;
+    sourceFiles = [];
+
+    constructor(disableWeaving) {
```

Comment:
if topFunction, platform and clock are mandatory, probably they should make part of the constructor (as the first parameters). Parameter disableWeaving could be optional and have a default value.

Reply:
Done. Changed the constructor to take 4 args, so that topFunction is mandatory, with the platform, target clock and weaving being optional, as per the previous discussion.